plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1094,6 +1109,30 @@ else if (avroSchemaLiteral != null) {
                     tableProperties.put(CSV_SEPARATOR_KEY, separator.toString());
                 });
 
+        // REGEX specific properties
+        getRegexPattern(tableMetadata.getProperties())
+                .ifPresentOrElse(
+                        regexPattern -> {
+                            checkFormatForProperty(hiveStorageFormat, HiveStorageFormat.REGEX, REGEX_PATTERN);
+                            try {
+                                Pattern.compile(regexPattern);
+                            }
+                            catch (PatternSyntaxException e) {
+                                throw new TrinoException(INVALID_TABLE_PROPERTY, format("Invalid REGEX pattern '%s' table property", regexPattern));
```

Comment:
This message looks similar to the one below, but is using a value and not the property name. How about
```java
throw new TrinoException(INVALID_TABLE_PROPERTY, "Invalid REGEX pattern value: " + regexPattern));

Reply:
Actually, it might be better to do the validation in the property metadata declaration. I'll comment there.

Reply:
To match the other table properties we decided to leave this as is, but it is something we can reconsider in the future.